### ISSUE TITLE
Feature: see report info from plot view

### ISSUE THREAD

**Issue by tsnowlan** [ewels/MegaQC#50]
Currently you can mouseover a point on a plot / trend and see some overall info about the report grouping. It would be really useful to be able to click on it and have it either take the user to a separate page with detailed info on that report, or pop up a sidebar with some additional information and a link to the more detailed report view. Maybe have an option to include a path to the MultiQC report with the MegaQC upload? I guess there are a few new features here. Thoughts?

**Comment by ewels** [comment]
Yes, definitely sounds like a good idea! Should be easy enough to have a custom JavaScript callback on the click event which triggers an API call with the sample ID.

Plot.ly tooltips are a pain to work with, so probably easiest if that new information is shown in a sidebar or [bootstrap modal](http://getbootstrap.com/docs/4.1/components/modal/) or something.

This also sort of ties into #29 a little where I'd like to make use of plot.ly's excellent [lasso selection tool](https://codepen.io/plotly/full/EPEGyE/) to make sample filters. It could be nice to make sample filters for a single sample here too, so that it's easy to view other graphs. Or perhaps that's overkill and we could instead make the plot URLs accept an optional parameter for a single sample / report ID. That would be cool.

**Comment by tsnowlan** [comment]
Filters on `sample_name` would definitely be good. We're actually going to be doing single sample report uploads to megaqc for a better visibility into which sample in a report is at a given data point.

To me the ideal drill down data flow (buzzwords AND alliteration!) would be:
1. Select 1 or more reports (directly or from a date range, or whatever method) for visualization
2. Mouse over / click / lasso select point or grouping of interest
3. Sidebar or modal pops up with more information about that grouping
  a. Option to replot with just the selected report
  b. Option (via mouseover, click box, something) to highlight a specific sample in new or original plot
4. View specific sample details from either the report detail modal or point on a new plot

That is a lot of work, but I think would make finding more out about a weird data point a lot easier. The individual sample level detail could also potentially bring back to the MultiQC report. Though storing those separate from the MegaQC installation would not be surprising, so that could be problematic.

**Comment by ewels** [comment]
Yes, I think this workflow would work really nicely - very similar to what I had in mind, but more refined and more in-page which would certainly be more user friendly. I think a plot sidebar would be perfect for this (we typically have plenty of page width available anyway).

> The individual sample level detail could also potentially bring back to the MultiQC report. Though storing those separate from the MegaQC installation would not be surprising, so that could be problematic.

Yes - we should have the original report location, but I imagine that most users (us, anyway) won't be able to do anything with this automatically as it's on a totally different server.

**Comment by ewels** [comment]
> We're actually going to be doing single sample report uploads to megaqc for a better visibility into which sample in a report is at a given data point.

I don't really understand what you mean by this...?

**Comment by tsnowlan** [comment]
On mouseover of a point in the trend view, the tooltip only gives one sample ID and its value from the overall report. It will always give the value / sample_id of the same point, which is not necessarily the point one is interested in. In the below case, the mouse was over the top point which has  `PCT_OFF_BAIT` > 0.25, but it was showing the ID and value for a point in the middle.

![megaqc_pcf_off_bait](https://user-images.githubusercontent.com/5105646/40778205-a8afc914-64d0-11e8-9b52-74a230a3d3d6.png)

This might be something that can be fixed in plotly, but a faster workaround to get things working here was just to import them singly so there isn't another data point that can be shown on the tooltip.

**Comment by ewels** [comment]
Hmm, this is definitely not the desired behaviour and should be fixed in the plot.ly code.

**Comment by ewels** [comment]
Ok, that particular behaviour is now fixed 👍
